### DISCUSSION TITLE
Custom taxa order plot_bar?

### DISCUSSION THREAD

**Discussion by jarrodscott** [ChiLiubio/microeco#45] [Q&A] [answered]
Hi everyone

Does anyone have a suggestion for specifying the order taxa appear in bar charts, heat maps, etc.? Following the example in the [tutorial](https://chiliubio.github.io/microeco/#trans_abund_class), it seems taxa are reverse ordered by total relative abundance. Is there a way to specify the order using a character vector? I have a mix of bacterial and archaeal phylum and I would like to list the bacteria first and then the archaea. 

`"Proteobacteria", "Acidobacteriota", "Actinobacteriota", "Bacteroidota", "Firmicutes", "Myxococcota", "Verrucomicrobiota", "Chloroflexi", "Planctomycetota", "Methylomirabilota", "Crenarchaeota", "Others"`

Here is a reproducible example from the tutorial

<img width="701" alt="Screen Shot 2021-07-09 at 12 31 25 PM" src="https://user-images.githubusercontent.com/16192182/125116060-c2fb1480-e0b1-11eb-8969-c10b58d248db.png">

```
data(sample_info_16S)
data(otu_table_16S)
data(taxonomy_table_16S)
data(phylo_tree_16S)
data(env_data_16S)

dataset <- microtable$new(sample_table = sample_info_16S, 
                                              otu_table = otu_table_16S, 
                                              tax_table = taxonomy_table_16S, 
                                              phylo_tree = phylo_tree_16S)
dataset$tidy_dataset()
print(dataset)

dataset$tax_table %<>% base::subset(Kingdom == "k__Archaea" | Kingdom == "k__Bacteria")
dataset$filter_pollution(taxa = c("mitochondria", "chloroplast"))
dataset$tidy_dataset()

dataset$cal_abund()
t1 <- trans_abund$new(dataset = dataset, taxrank = "Phylum", ntaxa = 10)
t1$plot_bar(others_color = "grey70", facet = "Group", xtext_keep = FALSE, legend_text_italic = FALSE)
```

**Comment by jarrodscott** [comment, marked as answer]
Yes! After hours of banging my head against the wall I figured it out. Of course, right after posting my question.

Here is how I reordered taxa names. Super simple once I understood how the data object is organized. 

The default code here organizes the plot like so...

```
tmp_me <- clone(ssu18_ps_work_me_taxrank)
tmp_me$plot_bar(others_color = "#323232", facet = "TEMP", 
                  xtext_keep = TRUE, xtext_type_hor = FALSE, 
                  legend_text_italic = FALSE, xtext_size = 6)
```
<img width="702" alt="Screen Shot 2021-07-09 at 1 08 46 PM" src="https://user-images.githubusercontent.com/16192182/125119593-f2f8e680-e0b6-11eb-84ee-14c1345578c7.png">

if I run this command with my custom taxa order:

```
tmp_me$use_taxanames <- c("Alphaproteobacteria", "Gammaproteobacteria", "Acidobacteriota", 
                              "Actinobacteriota", "Bacteroidota", "Firmicutes", "Myxococcota", 
                              "Verrucomicrobiota", "Chloroflexi", "Planctomycetota", 
                              "Methylomirabilota", "Crenarchaeota")
```

And then rerun the `plot_bar` function

```
tmp_me$plot_bar(others_color = "#323232", facet = "TEMP", 
                  xtext_keep = TRUE, xtext_type_hor = FALSE, 
                  legend_text_italic = FALSE, xtext_size = 6)
```

I get this:

<img width="699" alt="Screen Shot 2021-07-09 at 1 12 52 PM" src="https://user-images.githubusercontent.com/16192182/125119945-6569c680-e0b7-11eb-8bf7-466ece7677e9.png">

I love the `microeco` package. So intuitive. Thanks!

**Reply by ChiLiubio** [reply]
Cool example!

**Reply by ChiLiubio** [reply]
The input_taxaname parameter in trans_abund$new can also be used for this goal.

**Reply by ChiLiubio** [reply]
Now, the use_taxanames in the trans_abund object was renamed to data_taxanames.
